pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -259,6 +272,8 @@ type Controller struct {
 	// informerInit is set to true once the controller is running successfully. This ensures we do not
 	// return HasSynced=true before we are running
 	informerInit *atomic.Bool
+	// beginInitialSync is set to true when calling SyncAll.
```

Comment:
nit: can you make this clearer by adding "when calling SyncAll, but before SyncAll has completed"

Reply:
>but before SyncAll has completed

This is not right, do not modify it after SyncAll

Reply:
```go
func (cl *Client) SyncAll() {	
   cl.beginInitialSync.Store(true)
```

Seems to me like it says "we're in the process of syncing, but it may or may not have completed".

Reply:
changed to `beginSync`, i think it is more clear. Indicates the controller has begun syncing resources.